plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotWorkbenchActions.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2011 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.swtbot;
+
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.ProgressBar;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.eclipse.finder.matchers.WidgetMatcherFactory;
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.waits.ICondition;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotMenu;
+import org.hamcrest.Matcher;
+
+import java.util.List;
+
+/**
+ * SWTBot utility methods that perform general workbench actions.
+ */
+public final class SwtBotWorkbenchActions {
+
+  private static final int OPEN_PREFERENCES_DIALOG_DELAY_MS = 1000;
+
+  /**
+   * Opens the preferences dialog from the main Eclipse window.
+   * <p>
+   * Note: There are some platform-specific intricacies that this abstracts
+   * away.
+   */
+  public static void openPreferencesDialog(final SWTWorkbenchBot bot) {
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        if (SwtBotTestingUtilities.isMac()) {
+          // TODO: Mac has "Preferences..." under the "Eclipse" menu item.
+          // However,
+          // the "Eclipse" menu item is a system menu item (like the Apple menu
+          // item), and can't be reached via SWTBot.
+          openPreferencesDialogViaEvents(bot);
+        } else {
+          SWTBotMenu windowMenu = bot.menu("Window");
+          windowMenu.menu("Preferences").click();
+        }
+      }
+    });
+  }
+
+  /**
+   * Wait until all background tasks are complete.
+   */
+  public static void waitForIdle(SWTBot bot) {
+    while (!Job.getJobManager().isIdle()) {
+      bot.sleep(300);
+    }
+  }
+
+  /**
+   * Wait for the main shell progress bar to get removed.
+   */
+  public static void waitForMainShellProgressBarToFinish(final SWTWorkbenchBot bot) {
```

Comment:
No.  I've used `SwtBotWorkbenchActions.waitForIdle()` as it looks at all system and user jobs.  But I have verified it works changing `deleteProject()`
```
    SwtBotTestingUtilities.clickButtonAndWaitForWindowChange(bot, bot.button("OK"));
```
to
```
    bot.button("OK").click();
    SwtBotWorkbenchActions.waitForMainShellProgressBarToFinish(bot);
```
as the delete dialog contains a progress bar.